File: backend/src/main/java/org/akira/ladux/service/ColorService.java
package org.akira.ladux.service;

import org.akira.ladux.dto.catalog.request.ColorRequest;
import org.akira.ladux.dto.catalog.response.ColorResponse;
import org.akira.ladux.dto.common.PageResponse;
import org.akira.ladux.model.Color;
import org.springframework.data.domain.Pageable;

public interface ColorService {
    PageResponse<ColorResponse> getAllColors(Pageable pageable);

    Color addColor(ColorRequest request);

    Color updateColor(int id, ColorRequest request);

    void deleteColor(int id);
}

File: frontend/src/services/colorService.ts
import apiClient from './apiClient';
import { ColorResponse } from './productService';

export interface ColorRequest {
  name: string;
  hexCode: string;
}

export const colorService = {
  /**
   * Thêm màu sắc mới
   * POST /api/v1/admin/color
   */
  addColor: (data: ColorRequest): Promise<ColorResponse> => {
    return apiClient.post('/admin/color', data);
  },

  /**
   * Cập nhật màu sắc
   * PUT /api/v1/admin/color/{id}
   */
  updateColor: (id: number, data: ColorRequest): Promise<ColorResponse> => {
    return apiClient.put(`/admin/color/${id}`, data);
  },

  /**
   * Xóa màu sắc
   * DELETE /api/v1/admin/color/{id}
   */
  deleteColor: (id: number): Promise<void> => {
    return apiClient.delete(`/admin/color/${id}`);
  },
};
